### DISCUSSION TITLE
Error in the documentation on async guidance

### DISCUSSION THREAD

**Discussion by MartinHeimsoth-V** [fastapi/fastapi#11203] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# n/a
```

### Description

There seems to be a very confusing typo in the [documentation about using async](https://fastapi.tiangolo.com/async/#in-a-hurry). There it says:

> If your application (somehow) doesn't have to communicate with anything else and wait for it to respond, use `async def`.

And then in the next paragraph:

> If you just don't know, use normal `def`.

I have the impression that the `async` keyword should just be hop to the second paragraph to make sense.
![Screenshot 2024-02-26 at 17 31 50](https://github.com/tiangolo/fastapi/assets/97889291/4b44bb2e-9724-4b2b-912b-955e8a8bae79)

### Operating System

Other

### Operating System Details

Not relevant (just about documentation)

### FastAPI Version

0.110.0

### Pydantic Version

1.10.13

### Python Version

3.11

**Comment by YuriiMotov** [comment]
I agree that first statement is not clear:

> If your application (somehow) doesn't have to communicate with anything else and wait for it to respond, use `async def`.

Heavy calculations don't require communications with anything else, but they shouldn't be placed 'as is' into the `async def` endpoint.
So, I also think that it should be clarified.

But second statement is correct - if you are not sure whether you code is blocking or not, it's better to use `def` instead of `async def`
